### ISSUE TITLE
[3.4] Project config can't connect to database after install

### ISSUE THREAD

**Issue by janhenckens** [craftcms/cms#5697]
### Description
When starting a project from our [starter repo](https://github.com/statikbe/craft), we're running into the following error:

After installing, the setup script will try to apply the existing project.yaml but that throws the following error: `can't apply existing project config: Craft CMS can’t connect to the database with the credentials in config/db.php.`, eventhough seconds earlier it was able to install without any issues. 

```
    > add foreign key userpreferences_userId_fk: {{%userpreferences}} (userId) references {{%users}} (id) ... done (time: 0.028s)
    > add foreign key users_id_fk: {{%users}} (id) references {{%elements}} (id) ... done (time: 0.017s)
    > add foreign key users_photoId_fk: {{%users}} (photoId) references {{%assets}} (id) ... done (time: 0.021s)
    > add foreign key volumefolders_parentId_fk: {{%volumefolders}} (parentId) references {{%volumefolders}} (id) ... done (time: 0.019s)
    > add foreign key volumefolders_volumeId_fk: {{%volumefolders}} (volumeId) references {{%volumes}} (id) ... done (time: 0.028s)
    > add foreign key volumes_fieldLayoutId_fk: {{%volumes}} (fieldLayoutId) references {{%fieldlayouts}} (id) ... done (time: 0.025s)
    > add foreign key widgets_userId_fk: {{%widgets}} (userId) references {{%users}} (id) ... done (time: 0.018s)
    > populating the info table ... done
    > can't apply existing project config: Craft CMS can’t connect to the database with the credentials in config/db.php.
    > renaming project.yaml to project-20200208-083243.yaml and moving to config backup folder ... done
```

Testing this with the previous version of the repo (1.4.2, which ran Craft 3.3.19) it did work, the latest release (on Craft 3.4.7.1) has the error.

### Steps to reproduce

1. Run `composer create-project statikbe/craft test`
2. Run test/craft setup and install Craft

### Additional info

- Craft version: 3.4.7.1
- PHP version: 7.3.8
- Database driver & version: 5.7.28

**Comment by janhenckens** [comment]
Found the issue. `config/db.php` still had server, database and driver settings. Replacing those with the DSN fixed it.

Still strange that it worked for the rest of the install though....

**Comment by brandonkelly** [comment]
> Still strange that it worked for the rest of the install though....

I can see how that could possibly happen, if you started with the `./craft setup` command which sets all the DB connection preferences, and then answered `yes` to the `Install Craft now?` prompt, which would end up installing Craft using the same DB connection that the `setup` command already established, rather than going through `.env` + `config/db.php`.

**Comment by janhenckens** [comment]
@brandonkelly can I pick your brain on this one a second thime?

My `config/db.php` and `.env` match Craft's and still I'm getting the same error.

Running `composer create-project statikbe/craft test` should show give you the error. It still creates all tables and but errors out when creating the user
When I then delete the tables and run the setup again, it works. (probably because then it already had the `.env` & `config/db.php` values.

I can only reproduce it with our `project.yaml`, not with `craftcms/craft`.

Appreciate any help you could provide!

**Comment by brandonkelly** [comment]
@janhenckens I was able to reproduce using your starter repo. Issue ended up being due to another change in 3.4, which is that `craft\queue\Queue` now stores its own reference to the `db` component internally (to be more like `yii\queue\db\Queue`), and that wasn’t getting updated when the app’s `db` component got swapped out based on the updated connection settings.

Fixed for the next release, by reconfiguring the existing `db` component once the connection settings are defined, rather than swapping it out with a new one.
